### ISSUE TITLE
Enter should not submit forms when popup is opened

### ISSUE THREAD

**Issue by tapopov** [telerik/kendo-angular#390]
Let's take the following example:
```
@Component({
  selector: 'my-app',
  template: `
    <form #form="ngForm" (ngSubmit)="onSubmit($event)">
        <div class="example-config">
            The selected gender is <strong>{{ gender | json }}</strong>
        </div>
        <label>
            Select Gender:
            <kendo-combobox
                name="gender"
                [data]="genders"
                [textField]="'text'"
                [valueField]="'value'"
                [(ngModel)]="gender"
            >
            </kendo-combobox>
        </label>
    </form>
  `
})
class AppComponent {
    public genders: Array<{ text: string, value: number }> = [
        { text: "Male", value: 1 },
        { text: "Female", value: 2 },
        { text: "Other", value: 3 }
    ];

    public gender: { text: string, value: number } = { text: "Female", value: 2 };
  
    public onSubmit(event: any) {
     alert(); 
    }
}
```

## Current behavior:
If a ComboBox item is focused and `enter` key is pressed, the form will be submitted and `alert` will pop up.
## Expected behavior:
Pressing `enter` should *not* submit the form. Instead, the focused item should be selected and the ComboBox collapsed.
## Workaround:
Prevent the form's `keydown.enter` event default behavior:
```
    <form #form="ngForm" (ngSubmit)="onSubmit($event)" (keydown.enter)="$event.preventDefault()">
```

**Comment by Xizario** [comment]
It is [implicit form submission](https://www.w3.org/TR/html5/forms.html#implicit-submission), not caused by the ComboBox. The browser itself decides to submit the form, since there is only one input element inside it.

You may test these steps:
1. Add normal `<input/>` element inside your form (before or after the ComboBox, does not matter). What matters is they are 2 input elements now.
2. Load the page and press Enter into the ComboBox: as result the form is **not** submitted.
3. Now remove the ComboBox from your form, leave only the input element.
4. Load the page, press Enter into the input element: as result the form is submitted (yet there is no ComboBox).

**Comment by tapopov** [comment]
@Xizario, should we allow the browser to implicitly submit the form while the user is trying to select an item from the ComboBox's popup? I think not.

**Comment by Xizario** [comment]
@tapopov, I am sorry, I missed to read where the ComboBox was expanded. Now in case user select value from the list, first enter probably must collapse the list and selects the value, and the second enter to submit the form. Similar to how it work for native elements:
```
<form>
  <input list="browsers" name="browser">
  <datalist id="browsers">
    <option value="Internet Explorer">
    <option value="Firefox">
    <option value="Chrome">
    <option value="Opera">
    <option value="Safari">
  </datalist>
</form>
```

**Comment by tapopov** [comment]
Fixed in v0.39.7

**Comment by siyka-marinova** [comment]
https://github.com/telerik/kendo-angular-dropdowns/blob/master/e2e/combobox.spec.ts#L100
